src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -883,19 +915,19 @@ impl LinkCollector<'_, '_> {
                 path
             } else {
                 disambiguator = None;
-                &link
+                link
             }
-            .trim();
+            .trim()
+            .to_owned();
 
             if path_str.contains(|ch: char| !(ch.is_alphanumeric() || ch == ':' || ch == '_')) {
                 return;
             }
 
             // We stripped `()` and `!` when parsing the disambiguator.
             // Add them back to be displayed, but not prefix disambiguators.
-            link_text = disambiguator
-                .map(|d| d.display_for(path_str))
-                .unwrap_or_else(|| path_str.to_owned());
+            link_text =
+                disambiguator.map(|d| d.display_for(&path_str)).unwrap_or_else(|| path_str.clone());
```

Comment:
You forgot to remove this copy.

Reply:
Right now, path_str is a String, because the Key needs a String, so it was cloned here. Basically, it's a copy here or a copy later.

Reply:
Right, because the cache needs an owned value. Ok, makes sense.